rfc/2023/2023-05-01-va-rfc-lightweight-apps-paths.md

```diff
@@ -0,0 +1,96 @@
+
+VFS and Platform engineers should also look into what tools might help existing applications achieve better performance by shifting more logic to `vets-api` and HTML generation to build time. Frameworks like Next.js support [static HTML export](https://nextjs.org/docs/advanced-features/static-html-export), which might provide a migration path for existing applications.
+
+## Risks
+
+
+>List the risks of this approach
+> * There are always risks. What are the risks of this solution?
+> * These are the things people will bring up in opposition to your idea or plans. Acknowledge them.
+
+
+- There aren't enough guide rails without a framework like React.
+    - Platform and VFS teams should evaluate minimalist JS frameworks ( [alpinejs](https://alpinejs.dev), [htmx](https://htmx.org), etc.) for lightweight interactivity/dynamic content.
+    - Platform and VFS teams can collaborate on best practices and update the Platform documentation to reflect those best practices. Those best practice could also be incorporated into issue templates and collab cycle reviews.
+- SSG will require lots of additional tools, creating its own complexity
+    - `vets-website` already has support for ejs templates, and app entry points don't intrinsically have to use React, so we already have the tools we need
+    - Next.js comes up, and that's something that could be explored
+- We'll end up with "Not Invented Here" roll-your-own solutions
+    - Platform docs, standardized templates, and the collab cycle should mitigate this
+    - Various next-generation tools, e.g. Next.js, Svelte, Alpine.js, HTMX, could be evaluated and adopted
+
+## Alternatives
+
+> List the alternative approaches
+> * There are always alternatives. What other alternative solutions were considered? Not considered?
+> * What are their strengths, weaknesses, risks? Why weren’t they chosen?
+> * Do not allow bias of a solution to show in this section, ensure each alternative has been considering seriously or do not list it in this section
+
+Rather than adopt a lightweight apps path based on existing tools, we could immediately adopt a new framework that provides static page generation options, like Next.js. Adopting this across `vets-website` would be a large shift, so doing this incrementally for isolated applications might be a good approach. Taking this approach would require engineers to become familiar with the best practices and pitfalls of another framework. There may be a significant population of engineers with this knowledge who could help develop expertise across the Platform and VFS teams. It would still be important to measure and budget for performance. This approach probably requires further exploration and could not be immediately adopted.
```

Comment:
Re-word this so "allow for incremental adoption of a new framework" is the focus.

Reply:
I went looking for examples or documentation of an incremental migration of a React application to Next.js and found a [blog post on converting the BBC to Next](https://vercel.com/blog/migrating-a-large-open-source-react-application-to-next-js-and-vercel) (appears to be a proof of concept), and a link to https://nextjs.org/docs/migrating/incremental-adoption that redirects to the root docs page. So incremental adoption of Next.js seems possible, needs more research